framework/src/modules/interoperability/base_interoperability_cc_commands.ts

```diff
@@ -14,12 +14,18 @@
 
 import { BaseCCCommand } from './base_cc_command';
 import { BaseInteroperableAPI } from './base_interoperable_api';
+import { SidechainInteroperabilityStore } from './sidechain/store';
+import { getStoreFunction } from './types';
 
 export abstract class BaseInteroperabilityCCCommand extends BaseCCCommand {
-	protected readonly _interoperableCCAPIs = new Map<number, BaseInteroperableAPI>();
+	protected readonly interoperableCCAPIs = new Map<number, BaseInteroperableAPI>();
 
 	public constructor(moduleID: number, interoperableCCAPIs: Map<number, BaseInteroperableAPI>) {
 		super(moduleID);
-		this._interoperableCCAPIs = interoperableCCAPIs;
+		this.interoperableCCAPIs = interoperableCCAPIs;
+	}
+
+	protected getInteroperabilityStore(getStore: getStoreFunction) {
+		return new SidechainInteroperabilityStore(this.moduleID, getStore, this.interoperableCCAPIs);
```

Comment:
The name says `getInteroperabilityStore` but it's returning `SidechainInteroperabilityStore`. I think it's a bit misleading.

Reply:
What do you think about making this abstract to inform to type system that there is a method here, but implementing store constructors in classes that inherit `BaseInteroperabilityCCCommand`? Because I didn't understand why we implement sidechain interoperability store constructor for every subclass, don't we need to create also mainchain interoperability store in some cases?

Reply:
yes this was a mistake, while moving this function to BaseInteroperabilityStore I forgot they need specific Interoperability module's store. I will make this function abstract and to be implemented in their respective classes